apps/examples/smartfs_powercut/smartfs_powercut_main.c

```diff
@@ -171,76 +212,148 @@ static int create_file(char *path, ssize_t bufsize, file_type_t type)
 	return ret;
 }
 
-static int verify_random_file(char *path, int bufsize, int *index)
+static int verify_random_file_data(int fd, int bufsize)
+{
+	int len;
+	uint16_t crc;
+	int data_len = bufsize - sizeof(g_buf.crc);
+
+	len = read(fd, (void *)&g_buf, bufsize);
+	if (len != bufsize) {
+		/* If write is not finished properly */
+		printf("verify_random_file : datachunk is not written fully, ret : %d\n", len);
+		return ERROR;
+	} else {
+		/* Maybe this won't happen but check crc for double confirmation */
+		crc = crc16((const uint8_t *)g_buf.data, data_len);
+		if (crc != g_buf.crc) {
+			printf("verify_random_file : crc mismatch, calulcated crc : %d, written crc : %d\n", crc, g_buf.crc);
+			return ERROR;
+		}
+	}
+	return OK;
+}
+
+static int verify_text_file_data(int fd, char *sample_buf, uint16_t sample_crc, int bufsize)
```

Comment:
I don't think these 2 functions are tightly related to its type... very similar.

Reply:
One calculates CRC on data part of buffer -> Random file function
Other uses CRC calculated on entire buffer of sample_file.txt and entire buffer read from file -> Text file function

Reply:
Of course I know it but I think there's no reason for divided functions.
please consider comment what I commented in line 281

Reply:
updated